cmd/akashctl/main.go

```diff
@@ -41,8 +40,28 @@ func main() {
 
 	// Add --chain-id to persistent flags and mark it required
 	root.PersistentFlags().String(flags.FlagChainID, "", "Chain ID of tendermint node")
+	root.PersistentFlags().String(flags.FlagNode, "", "The node address")
+
 	root.PersistentPreRunE = func(_ *cobra.Command, _ []string) error {
-		return initConfig(root)
+		// viper bindings below should be applied to root command rather then
+		// to an argument from this function. otherwise viper won't able to find them
+		if err := viper.BindPFlag(flags.FlagChainID, root.PersistentFlags().Lookup(flags.FlagChainID)); err != nil {
+			return err
+		}
+
+		if err := viper.BindPFlag(cli.EncodingFlag, root.PersistentFlags().Lookup(cli.EncodingFlag)); err != nil {
+			return err
+		}
+
+		if err := viper.BindPFlag(cli.OutputFlag, root.PersistentFlags().Lookup(cli.OutputFlag)); err != nil {
+			return err
+		}
+
+		if err := viper.BindPFlag(flags.FlagNode, root.Flags().Lookup(flags.FlagNode)); err != nil {
```

Comment:
are all of these flags always present?  `akashctl keys list` for instance might not take `--node`.

It's fine to have here so long as it doesn't crash when `root.Flags().Lookup(flags.FlagNode)` returns `nil`

Reply:
it does indeed